solana/program_v2/programs/gateway_v2/src/util.rs

```diff
@@ -47,15 +54,47 @@ pub fn calculate_network_and_gatekeeper_fee(fee: u64, split: u16) -> (u64, u64)
     (network_fee as u64, gatekeeper_fee as u64)
 }
 
+pub fn create_and_invoke_transfer<'a>(
+    spl_token_address: Program<'a, Token>,
+    source_account: Account<'a, TokenAccount>,
+    destination_account: Account<'a, TokenAccount>,
+    authority_account: Signer<'a>,
+    signer_pubkeys: &[&Pubkey],
+    amount: u64,
+) -> ProgramResult {
+    let transfer_instruction_network_result = transfer(
+        &spl_token_address.key(),
+        &source_account.key(),
+        &destination_account.key(),
+        &authority_account.key(),
+        signer_pubkeys,
+        amount,
+    );
+    let instruction = match transfer_instruction_network_result {
+        Ok(instruction) => instruction,
+        Err(error) => panic!("Transfer failed: {:?}", error),
```

Comment:
What should I do instead of panic?

Reply:
@rado0x54 what do you think?

Reply:
Always return an Anchor Result and bubble this all the way up. Create a custom anchor error for that case. (if there is not already a matching one in the Anchor library.

Reply:
Thanks! I like this way of handling errors :)

Reply:
Well I feel it's the rust way. Lost of `?` usage. Makes for clean code.